pkg/apis/internal.acorn.io/v1/memory.go

```diff
@@ -0,0 +1,33 @@
+package v1
+
+import (
+	"cuelang.org/go/cue/literal"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func ParseMemoryMap(memory map[string]string) (map[string]int64, error) {
+	resources := map[string]int64{}
+	for key, value := range memory {
+		quantity, err := ParseQuantityString(value)
+		if err != nil {
+			return nil, err
+		}
+		resources[key] = quantity
+	}
+	return resources, nil
+}
+
+func ParseQuantityString(memory string) (int64, error) {
+	numInfo := literal.NumInfo{}
+	err := literal.ParseNum(memory, &numInfo)
+	if err != nil {
+		return -1, err
+	}
+
+	quantity, err := resource.ParseQuantity(numInfo.String())
```

Comment:
Cue actually uses `"github.com/cockroachdb/apd/v2.Decimal"` under the hood for its parsing. My concern with using `Quantity` here is that there may be a difference with the way the two parse the `numInfo.String()` value.

Would there be a problem with using `numInfo.Decimal` here instead?

Reply:
After huddle conversation: use `strconv.ParseInt` here to get an `int64`.